docs/pages/architecture/agents.mdx

```diff
@@ -0,0 +1,255 @@
+resources:
+
+|Resource kind|How users obtain certificates|
+|---|---|
+|Databases|The user retrieves a certificate for a database with `tsh db login`. The Teleport Database Service verifies the certificate when a user connects to a database.|
+|Kubernetes clusters|The user runs `tsh kube login`, which updates the local kubeconfig to contain a certificate signed by the Teleport user CA. The user can then execute `kubectl` commands against Teleport-protected Kubernetes clusters.|
+|Servers|After running `tsh login`, the user receives a certificate that contains authorization information for servers. The SSH Service checks the certificate when a user authenticates to a Teleport-protected server.|
+|Web applications|The user authenticates to the Teleport Web UI and visits a Teleport-protected application. The Teleport Proxy Service forwards traffic to the application along with Teleport-signed JSON web tokens that the application can verify. |
+|Windows desktops|Users authenticate to the Teleport Web UI. The Web UI initiates a WebSocket session with the Teleport Desktop Service and communicates with the service using Teleport Desktop Protocol (TDP). The Desktop Service converts TDP traffic to Remote Desktop Protocol traffic.|
+
+In most cases (except for some self-hosted clusters), the user will have
+connected to the Auth Service through the Teleport Proxy Service. The Auth
+Service forwards user certificates through the Proxy Service over mutual TLS.  
+
+## Establishing reverse tunnels
+
+In most cases, when an agent joins a Teleport cluster, it establishes a reverse
+SSH tunnel to the Proxy Service. As such, users do not need to have direct
+connectivity to agents or the resources they are connected to. As long as an
+agent can dial back to the cluster's Proxy Service, it can be located behind a
+firewall.
+
+![Tunnel Mode](../../img/architecture/k8s-tunnel.png)
+
+The connection between the Proxy Service and a Teleport agent instance is also
+authenticated with mutual TLS. The Proxy Service generates a short-lived X.509
+certificate signed by the cluster's host CA, with the client's identity and
+routing information for an infrastructure resource encoded in it, and uses it to
+authenticate with the Teleport agent.
+
+The Proxy Service authenticates the connection and dispatches it to the
+appropriate agent instance based on the routing information encoded in the
+client certificate, over the reverse tunnel.
+
+<Details title="Connecting to agents without reverse tunnels">
+
+It is possible to join Teleport agents to a cluster [through the Teleport Auth
+Service](../agents/join-services-to-your-cluster/join-token.mdx#connecting-directly-to-the-auth-service.mdx).
+Once an agent joins a cluster through the Teleport Auth Service, the Teleport
+Proxy Service dials the agent directly, without creating a reverse tunnel. This
+mode supports the Teleport SSH Service, Teleport Desktop Service, and Teleport
+Kubernetes Service.
```

Comment:
If we were to include discovery service, it doesn't require reverse tunnel either